CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.1.3 FATAL_ERROR)
```

Comment:
> nobody is on cmake <3 anymore

You will be surprised.

Although, define "legacy". It is not deprecated, and it is not recommended by CMake developers not to use.

Effectively, without it we only fail to give one good error message, in case if someone _accidentally_ tries to use a very old CMake.

Reply:
None of the other functions in the CMake file would work if they were on <3. That's why you don't see it in modern CMake configs anymore.

Reply:
> None of the other functions in the CMake file would work if they were on <3

That is incorrect. Strictly speaking: "_some_ of the functions in this build tree will not work, _or will have a different outcome, that the script author intended_"